scripts/o3de/o3de/repo_properties.py

```diff
@@ -263,6 +264,32 @@ def _edit_objects(object_typename:str,
 
     repo_json[f'{object_typename}s_data'] = repo_objects_data
 
+def _auto_update_json(object_type: str or list,
+                      repo_path: pathlib.Path = None,
+                      repo_json: dict = None,
+                      ):
+    
+    repo_directory = os.path.dirname(repo_path)
+    expected_files = {"project.json": [],
+                      "gem.json": [],
+                      "template.json": []}
+    for directory, sub_directory, files, in os.walk(repo_directory):
+        current_depth = directory[len(repo_directory):].count(os.sep)
+        for filename in files:
+            if filename in expected_files.keys() and current_depth < 3:
+                file_path = pathlib.Path(directory)
+                expected_files[filename].append(file_path)
+    
+    objects = object_type.split() if isinstance(object_type, str) else object_type
+    for object_name in objects:
+        if object_name == 'gem':
```

Comment:
if you do `.lower()` to `object_name`, then the auto_update list doesn't have to be case sensitive and you can update documentation for the argument.
```suggestion
        if object_name.lower() == 'gem':
```

Reply:
This makes the function a lot more robust 👍